### PR TITLE
Play page code seperation

### PR DESCRIPTION
**Pull request by kdhaka94**
This is to separate the code on the Play page for the levels and basically separate the conditions for play mode and test mode separately, and create a different component for those

### CODE REVIEW

**Comment by webdevcody** [comment]
useful refactoring!  I will say, try to keep the components co-located in the pages that use them.  The play and test mode are not used on any other page, so maybe put them directly in the page.tsx route they are used in

**Comment by delasy** [comment]
@kdhaka94 can you resolve conflicts with main branch? thanks

**Comment by kdhaka94** [comment]
I wasn't able to test the latest due to convex issue, figuring it out, I tried to put in most of the logic and not mess it up, Please do let me know if any issue occurs, and also when pulled in the latest it seems somebody forgot to typecast the handlers and map.ts

**Comment by kdhaka94** [comment]
> useful refactoring! I will say, try to keep the components co-located in the pages that use them. The play and test mode are not used on any other page, so maybe put them directly in the page.tsx route they are used in

Yeah, I do get your point but I feel that the pages folder itself gets too much cluttered if not handled properly and the [atomic design concept](https://medium.com/@janelle.wg/atomic-design-pattern-how-to-structure-your-react-application-2bb4d9ca5f97) might be a better solution to and if there is a case where the number of developers is exceeding, then using [plopJs](https://plopjs.com/) creates a much more developer friendly environment, I can go on and on but yeah just my opinion 😁

File: app/play/[level]/page.tsx [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+import { Visualizer } from "@/components/Visualizer";
+import { Map } from "@/components/Map";
+import Link from "next/link";
+import { ChevronLeftIcon } from "@radix-ui/react-icons";
 import { Tabs, TabsList, TabsTrigger } from "@/components/ui/tabs";
-import { api } from "@/convex/_generated/api";
+import TestMode from "./test-mode";
+import { handleCellClick, handlePlacementModeChange } from "@/utils/mapUtils";
+import { useRouter } from "next/navigation";
 import { ZombieSurvival } from "@/simulators/zombie-survival";
```
Comment (delasy) [inline app/play/[level]/page.tsx:15]: prettier plugin not used

File: app/play/[level]/page.tsx [hunk 2]
```diff
@@ -83,47 +63,18 @@ export default function PlayLevelPage({
...
-        newMap[y][x] = " ";
-        setBlockCount(blockCount - 1);
-      } else if (blockCount < 2) {
-        if (newMap[y][x] === " ") {
-          newMap[y][x] = "B";
-          setBlockCount(blockCount + 1);
-        }
-      }
-    }
-    setPlayerMap(newMap);
```
Comment (delasy) [inline app/play/[level]/page.tsx:115]: doesn't make much sense moving this functionality to a separate file, I don't think we will use it in any other part of the project

File: convex/maps.ts [hunk 3]
```diff
@@ -154,13 +154,13 @@ export const publishMap = adminMutationBuilder({
   args: {
     map: v.array(v.array(v.string())),
   },
-  handler: async (ctx, args) => {
+  handler: async (ctx: any, args: any) => {
```
Comment (delasy) [inline convex/maps.ts:157]: I think you shouldn't be replacing deduced types with `any`

File: convex/maps.ts [hunk 4]
```diff
@@ -154,13 +154,13 @@ export const publishMap = adminMutationBuilder({
...
     const maps = await ctx.db
       .query("maps")
-      .filter((q) => q.neq("level", undefined))
+      .filter((q: any) => q.neq("level", undefined))
```
Comment (delasy) [inline convex/maps.ts:160]: same here

File: convex/maps.ts [hunk 5]
```diff
@@ -154,13 +154,13 @@ export const publishMap = adminMutationBuilder({
...
       .collect();
 
-    const lastLevel = maps.sort((a, b) => b.level! - a.level!)[0].level!;
+    const lastLevel = maps.sort((a: any, b: any) => b.level! - a.level!)[0].level!;
```
Comment (delasy) [inline convex/maps.ts:163]: same here

File: convex/maps.ts [hunk 6]
```diff
@@ -234,10 +234,10 @@ export const approveMap = adminMutationBuilder({
   args: {
     mapId: v.id("maps"),
   },
-  handler: async (ctx, args) => {
+  handler: async (ctx: any, args: any) => {
```
Comment (delasy) [inline convex/maps.ts:237]: same here

File: convex/maps.ts [hunk 7]
```diff
@@ -234,10 +234,10 @@ export const approveMap = adminMutationBuilder({
...
     const maps = await ctx.db.query("maps").collect();
 
-    const lastLevel = maps.reduce((acc, map) => {
+    const lastLevel = maps.reduce((acc: number, map: any) => {
```
Comment (delasy) [inline convex/maps.ts:240]: same here

**Comment by delasy** [comment]
Hey @kdhaka94, can you check review and also resolve merge commits?

**Comment by delasy** [comment]
closing as stale, feel free to reopen once you fix all the merge conflicts!

**Pull request closed without merging**